From 5133f095bea2d147e86b99eacaa48363d08db35b Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@theqtcompany.com>
Date: Fri, 10 Jun 2016 14:46:24 +0200
Subject: [PATCH] Enable -fno_delete_null_pointer_checks on V8 for G++ 6

Detect g++ 6 and disable null pointer check optimizations on v8.

Change-Id: I5064823af3784786d455ce86592b5e65c1020f21
Task-number: QTBUG-53956
---

diff --git a/src/core/config/linux.pri b/src/core/config/linux.pri
index 8854a4b..b579e2a 100644
--- a/src/core/config/linux.pri
+++ b/src/core/config/linux.pri
@@ -29,6 +29,8 @@ use?(nss) {
         use_openssl_certs=1
 }
 
+gcc:!clang: greaterThan(QT_GCC_MAJOR_VERSION, 5): GYP_CONFIG += v8_no_delete_null_pointer_checks=1
+
 contains(QT_CONFIG, system-zlib): use?(system_minizip): GYP_CONFIG += use_system_zlib=1
 contains(QT_CONFIG, system-png): GYP_CONFIG += use_system_libpng=1
 contains(QT_CONFIG, system-jpeg): GYP_CONFIG += use_system_libjpeg=1
-- 
2.8.3

